xgi/classes/function.py

```diff
@@ -424,13 +425,13 @@ def set_node_attributes(H, values, name=None):
     """
     # Set node attributes based on type of `values`
     if name is not None:  # `values` must not be a dict of dict
-        try:  # `values` is a dict
+        if isinstance(values, dict):  # `values` is a dict
             for n, v in values.items():
                 try:
                     H._node_attr[n][name] = v
                 except KeyError:
-                    pass
-        except AttributeError:  # `values` is a constant
+                    warn(f"Node {n} does not exist!")
```

Comment:
Question: I'm confused as to when this ever gets executed - I thought the IDDict should catch all instances of missing node IDs?

Reply:
Good point. Fixed.